[Chooi Hao Wen] iP

hwc0419 opened the pull request.

hwc0419 committed:
Update skeleton code

hwc0419 committed:
Implement echo

hwc0419 committed:
Implement list

hwc0419 committed:
Add checkbox functionality to list

hwc0419 committed:
Tweak code to comply with CS2113 coding standards

hwc0419 committed:
Add ToDos, Events, Deadlines classes

yuhengr commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+        int MAX_NUMBERED_LIST_LENGTH = 100;
+        Task[] numberedList = new Task[MAX_NUMBERED_LIST_LENGTH];
+        int listCount = 0;
+
```
Perhaps it will help to move the different cases into methods so that it will be more readable

yuhengr reviewed (commented):
Overall good use of OOP to abstract the different classes. Could consider moving certain parts into methods to better improve readability

hwc0419 committed:
Improve Code Quality

hwc0419 committed:
Tweak code to comply with coding standards + More refactoring

hwc0419 committed:
Add Formatter Class to handle visual elements of the Interface
Also fixed a bug where the commands mark and unmark would crash for inputs where consecutive digits are wrapped by non digits.
This was due to the wrong regex expression used, causing SyntaxChecker to behave unexpectedly.

hwc0419 committed:
Customise visual elements in UI + More code refactoring

hwc0419 committed:
Handle errors

hwc0419 committed:
Fix bug for todo command

hwc0419 committed:
Add delete functionality

hwc0419 committed:
Refactor code

Jai2501 commented on src/main/java/BadTokenException.java:
```diff
@@ -0,0 +1,3 @@
+public class BadTokenException extends Exception{
+
+}
```
Empty Class. Please add at least a constructor with valid instantiations/assignments, so if this error is thrown, the description shown is correct.

Else please add a comment explaining why is it empty.

Jai2501 commented on src/main/java/CommandExecutor.java:
```diff
@@ -0,0 +1,59 @@
+    static CommandParser readUserCommand;
+    public static void beginListening() {
+        userInput = in.nextLine();
+    }
```
Add a line break between the variable declaration and start of function. It helps to improve readability.

```Java
static CommandParser readUserCommand;
    
public static void beginListening() {
        userInput = in.nextLine();
    }
```

Jai2501 commented on src/main/java/CommandExecutor.java:
```diff
@@ -0,0 +1,59 @@
+    public static void beginListening() {
+        userInput = in.nextLine();
+    }
+    public static void processInput() throws ProcessInputException{
```
Similarly here, add a line break.

Jai2501 commented on src/main/java/CommandExecutor.java:
```diff
@@ -0,0 +1,59 @@
+
+    public static void executeCommand() {
+            CommandList selectedCommand = CommandList.valueOf(readUserCommand.getCommandName());
+            switch (selectedCommand) {
```
Similarly here, add a line break.

This has been observed at multiple places in multiple files.

Jai2501 commented on src/main/java/CommandList.java:
```diff
@@ -0,0 +1,84 @@
+public enum CommandList {
```
Whats the rationale behind using `enum` vs `class` for CommandList?

Jai2501 commented on src/main/java/CommandList.java:
```diff
@@ -0,0 +1,84 @@
+    BYE, LIST, MARK, UNMARK, DELETE, TODO, DEADLINE, EVENT;
+    private static final String MAX_DIGIT_LENGTH_3_REGEX = "^[0-9]{1,3}$";
+    private static final String UNRESTRICTED_CHAR_LENGTH_REGEX = ".+";
+    private static final String[][] lutRegexSequence = {
```
`lutRegexSequence` is a constant. Please follow the correct naming convention.

I am assuming `lut` is `Lookup Table for Regular Expression Sequences`, and is generated with the help of GenAI? if so, please mention that you used it 😄

Jai2501 commented on src/main/java/CommandParser.java:
```diff
@@ -0,0 +1,50 @@
+    private int argumentCount = 0;
+    private String[] argumentTokens = {};
+    private String separator;
+    public CommandParser(String userInput) throws IllegalCommandException, ArgumentMismatchException, BadTokenException{
```
In the below, there is no space between `BadTokenException{`. Add a space to make it like `BadTokenException {`. This improves the code formatting.

```Java
public CommandParser(String userInput) throws IllegalCommandException, ArgumentMismatchException, BadTokenException {
```

Jai2501 commented on src/main/java/CommandParser.java:
```diff
@@ -0,0 +1,50 @@
+    public CommandParser(String userInput) throws IllegalCommandException, ArgumentMismatchException, BadTokenException{
+        String commandToken = userInput.split(" ", 2)[0];
+        String otherToken;
+        if (!SyntaxChecker.validateCommandToken(commandToken)) {
```
Lay out the code so that it adheres to the logical structure. Add a line break. I have observed this at multiple places in different files.

```Java
String otherToken;

if (!SyntaxChecker.validateCommandToken(commandToken)) {
```

Jai2501 commented on src/main/java/Formatter.java:
```diff
@@ -0,0 +1,103 @@
+public class Formatter {
+    final static String botName = "Howard Smith";
+    final static String emojiGrinning = "\uD83D\uDE00";
+    final static String emojiConfused = "\uD83D\uDE35\u200D\uD83D\uDCAB";
```
Please add the access specifier `public` or `private`. Also follow the constant naming convention.

It is also a convention to write as `<access_modifier> final <data_type> VARIABLE_NAME`

Example:
```Java
public static final String BOT_NAME = "Howard Smith";
```

Jai2501 commented on src/main/java/IllegalCommandException.java:
```diff
@@ -0,0 +1,2 @@
+public class IllegalCommandException extends Exception{
+}
```
Empty Class. Please add at least a constructor with valid instantiations/assignments, so if this error is thrown, the description shown is correct.

Else please add a comment explaining why is it empty.

Jai2501 commented on src/main/java/ProcessInputException.java:
```diff
@@ -0,0 +1,2 @@
+public class ProcessInputException extends Exception{
+}
```
Empty Class. Please add at least a constructor with valid instantiations/assignments, so if this error is thrown, the description shown is correct.

Else please add a comment explaining why is it empty.

Jai2501 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+public class Task {
```
Should this be made an Abstract Class? You may think about it, and get back to me if anything is unclear.

Jai2501 reviewed (commented):
Hi Hao Wen!

Good job on the work done so far!

I have reviewed your code and have left some comments. Please go through them and get back to me for any clarifications.

Thanks!

hwc0419 committed:
Refactor code

hwc0419 committed:
Refactor code

hwc0419 committed:
Add save user data functionality

hwc0419 committed:
Add find functionality

hwc0419 committed:
Add javadoc comments

hwc0419 committed:
Merge pull request #1 from hwc0419/branch-Level-9

Add find functionality

hwc0419 committed:
Merge branch 'master' into A-JavaDoc

hwc0419 committed:
Merge pull request #2 from hwc0419/A-JavaDoc

Add javadoc comments

hwc0419 committed:
Add user guide

hwc0419 committed:
Update README.md

hwc0419 committed:
Update README.md

hwc0419 committed:
Fix formatting issues

hwc0419 committed:
Add explanations for empty error classes + reorder syntax to fit naming convention
